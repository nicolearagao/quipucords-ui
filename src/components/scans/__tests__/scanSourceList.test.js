import React from 'react';
import { ScanSourceList } from '../scanSourceList';

describe('ScanSourceList Component', () => {
  it('should render a basic component', () => {
    const props = {
      id: 1,
      scanJobList: [
        {
          connectTaskStatus: 'lorem',
          connectTaskStatusMessage: 'ipsum',
          id: 1,
          inspectTaskStatus: 'dolor',
          inspectTaskStatusMessage: 'sit',
          name: 'test',
          sourceType: ''
        }
      ]
    };

    const component = renderComponent(<ScanSourceList {...props} />);

    expect(component).toMatchSnapshot('basic');
  });

  it('should render a basic component error', () => {
    const props = {
      id: 1,
      error: true,
      errorMessage: 'Lorem Ipsum.'
    };

    const component = renderComponent(<ScanSourceList {...props} />);
    expect(component).toMatchSnapshot('error');
  });

  it('should render a basic component pending', () => {
    const props = {
      id: 1,
      pending: true
    };

    const component = renderComponent(<ScanSourceList {...props} />);
    expect(component).toMatchSnapshot('pending');
  });

  it('should handle multiple status messages', () => {
    const props = {
      id: 1,
      scanJobList: [
        {
          connectTaskStatus: 'pending',
          connectTaskStatusMessage: 'ipsum',
          id: 1,
          inspectTaskStatus: 'dolor',
          inspectTaskStatusMessage: 'sit',
          name: 'test',
          sourceType: ''
        }
      ]
    };

    const component = renderComponent(<ScanSourceList {...props} />);
    expect(component).toMatchSnapshot('connect status');

    props.scanJobList[0].connectTaskStatus = 'completed';
    const componentInspect = renderComponent(<ScanSourceList {...props} />);
    expect(componentInspect).toMatchSnapshot('inspect status');

    props.scanJobList[0].inspectTaskStatus = undefined;
    const componentFallback = renderComponent(<ScanSourceList {...props} />);
    expect(componentFallback).toMatchSnapshot('fallback status');
  });
});
