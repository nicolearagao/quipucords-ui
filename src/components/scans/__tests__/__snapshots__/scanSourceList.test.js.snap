// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScanSourceList Component should handle multiple status messages: connect status 1`] = `
<scansourcelist
  props="{
  "id": 1,
  "scanJobList": [
    {
      "connectTaskStatus": "pending",
      "connectTaskStatusMessage": "ipsum",
      "id": 1,
      "inspectTaskStatus": "dolor",
      "inspectTaskStatusMessage": "sit",
      "name": "test",
      "sourceType": ""
    }
  ],
  "error": false,
  "errorMessage": null,
  "pending": false,
  "children": []
}"
>
  <ul
    class="pf-c-list pf-m-plain"
  >
    <li
      class=""
    >
      <ul
        class="pf-c-list pf-m-plain"
      >
        <li
          class=""
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 1024 1024"
            width="1em"
          >
            <path
              d="M943.3,317.2 L706.9,80.7 C653,26.9 582.5,0 512,0 C441.5,0 371,26.9 317.1,80.7 L80.7,317.1 C-26.9,424.7 -26.9,599.2 80.7,706.8 L317.1,943.2 C371,997.1 441.5,1024 512,1024 C582.5,1024 653,997.1 706.9,943.3 L943.3,706.9 C1050.9,599.3 1050.9,424.8 943.3,317.2 Z M866.4,602.2 L602.8,866.1 C552.8,916.1 471.8,916.2 421.7,866.2 L157.7,602.2 C107.8,552.2 107.8,471.2 157.8,421.3 L421.4,158.4 C471.3,108.6 552.1,108.6 602.1,158.3 L866.3,421.3 C916.3,471.3 916.3,552.2 866.4,602.2 Z M521.3,576 C627.5,576 713.7,502 713.7,413.7 C713.7,325.4 627.6,253.6 521.3,253.6 C366,253.6 334.5,337.7 329.2,407.2 C329.2,414.3 335.2,416 343.5,416 L445,416 C450.5,416 458,415.5 460.8,406.5 C460.8,362.6 582.9,357.1 582.9,413.6 C582.9,441.9 556.2,470.9 521.3,473 C486.4,475.1 447.3,479.8 447.3,521.7 L447.3,553.8 C447.3,570.8 456.1,576 472,576 C488,576.1 521.3,576 521.3,576 L521.3,576 Z M575.3,751.3 L575.3,655.3 C575.3,650.9 573.7,647.3 570.6,644 C567.5,640.9 563.7,639.3 559.3,639.3 L463.3,639.3 C458.9,639.3 455.3,640.9 452,644 C448.9,647.1 447.3,650.9 447.3,655.3 L447.3,751.3 C447.3,755.7 448.9,759.3 452,762.6 C455.1,765.7 458.9,767.3 463.3,767.3 L559.3,767.3 C563.7,767.3 567.3,765.7 570.6,762.6 C573.8,759.4 575.3,755.6 575.3,751.3 L575.3,751.3 Z"
            />
          </svg>
           
          test
        </li>
        <li
          class=""
        >
          Connection Scan: ipsum
        </li>
      </ul>
    </li>
  </ul>
</scansourcelist>
`;

exports[`ScanSourceList Component should handle multiple status messages: fallback status 1`] = `
<scansourcelist
  props="{
  "id": 1,
  "scanJobList": [
    {
      "connectTaskStatus": "completed",
      "connectTaskStatusMessage": "ipsum",
      "id": 1,
      "inspectTaskStatusMessage": "sit",
      "name": "test",
      "sourceType": ""
    }
  ],
  "error": false,
  "errorMessage": null,
  "pending": false,
  "children": []
}"
>
  <ul
    class="pf-c-list pf-m-plain"
  >
    <li
      class=""
    >
      <ul
        class="pf-c-list pf-m-plain"
      >
        <li
          class=""
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 1024 1024"
            width="1em"
          >
            <path
              d="M943.3,317.2 L706.9,80.7 C653,26.9 582.5,0 512,0 C441.5,0 371,26.9 317.1,80.7 L80.7,317.1 C-26.9,424.7 -26.9,599.2 80.7,706.8 L317.1,943.2 C371,997.1 441.5,1024 512,1024 C582.5,1024 653,997.1 706.9,943.3 L943.3,706.9 C1050.9,599.3 1050.9,424.8 943.3,317.2 Z M866.4,602.2 L602.8,866.1 C552.8,916.1 471.8,916.2 421.7,866.2 L157.7,602.2 C107.8,552.2 107.8,471.2 157.8,421.3 L421.4,158.4 C471.3,108.6 552.1,108.6 602.1,158.3 L866.3,421.3 C916.3,471.3 916.3,552.2 866.4,602.2 Z M521.3,576 C627.5,576 713.7,502 713.7,413.7 C713.7,325.4 627.6,253.6 521.3,253.6 C366,253.6 334.5,337.7 329.2,407.2 C329.2,414.3 335.2,416 343.5,416 L445,416 C450.5,416 458,415.5 460.8,406.5 C460.8,362.6 582.9,357.1 582.9,413.6 C582.9,441.9 556.2,470.9 521.3,473 C486.4,475.1 447.3,479.8 447.3,521.7 L447.3,553.8 C447.3,570.8 456.1,576 472,576 C488,576.1 521.3,576 521.3,576 L521.3,576 Z M575.3,751.3 L575.3,655.3 C575.3,650.9 573.7,647.3 570.6,644 C567.5,640.9 563.7,639.3 559.3,639.3 L463.3,639.3 C458.9,639.3 455.3,640.9 452,644 C448.9,647.1 447.3,650.9 447.3,655.3 L447.3,751.3 C447.3,755.7 448.9,759.3 452,762.6 C455.1,765.7 458.9,767.3 463.3,767.3 L559.3,767.3 C563.7,767.3 567.3,765.7 570.6,762.6 C573.8,759.4 575.3,755.6 575.3,751.3 L575.3,751.3 Z"
            />
          </svg>
           
          test
        </li>
        <li
          class=""
        >
          Connection Scan: ipsum
        </li>
      </ul>
    </li>
  </ul>
</scansourcelist>
`;

exports[`ScanSourceList Component should handle multiple status messages: inspect status 1`] = `
<scansourcelist
  props="{
  "id": 1,
  "scanJobList": [
    {
      "connectTaskStatus": "completed",
      "connectTaskStatusMessage": "ipsum",
      "id": 1,
      "inspectTaskStatus": "dolor",
      "inspectTaskStatusMessage": "sit",
      "name": "test",
      "sourceType": ""
    }
  ],
  "error": false,
  "errorMessage": null,
  "pending": false,
  "children": []
}"
>
  <ul
    class="pf-c-list pf-m-plain"
  >
    <li
      class=""
    >
      <ul
        class="pf-c-list pf-m-plain"
      >
        <li
          class=""
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 1024 1024"
            width="1em"
          >
            <path
              d="M943.3,317.2 L706.9,80.7 C653,26.9 582.5,0 512,0 C441.5,0 371,26.9 317.1,80.7 L80.7,317.1 C-26.9,424.7 -26.9,599.2 80.7,706.8 L317.1,943.2 C371,997.1 441.5,1024 512,1024 C582.5,1024 653,997.1 706.9,943.3 L943.3,706.9 C1050.9,599.3 1050.9,424.8 943.3,317.2 Z M866.4,602.2 L602.8,866.1 C552.8,916.1 471.8,916.2 421.7,866.2 L157.7,602.2 C107.8,552.2 107.8,471.2 157.8,421.3 L421.4,158.4 C471.3,108.6 552.1,108.6 602.1,158.3 L866.3,421.3 C916.3,471.3 916.3,552.2 866.4,602.2 Z M521.3,576 C627.5,576 713.7,502 713.7,413.7 C713.7,325.4 627.6,253.6 521.3,253.6 C366,253.6 334.5,337.7 329.2,407.2 C329.2,414.3 335.2,416 343.5,416 L445,416 C450.5,416 458,415.5 460.8,406.5 C460.8,362.6 582.9,357.1 582.9,413.6 C582.9,441.9 556.2,470.9 521.3,473 C486.4,475.1 447.3,479.8 447.3,521.7 L447.3,553.8 C447.3,570.8 456.1,576 472,576 C488,576.1 521.3,576 521.3,576 L521.3,576 Z M575.3,751.3 L575.3,655.3 C575.3,650.9 573.7,647.3 570.6,644 C567.5,640.9 563.7,639.3 559.3,639.3 L463.3,639.3 C458.9,639.3 455.3,640.9 452,644 C448.9,647.1 447.3,650.9 447.3,655.3 L447.3,751.3 C447.3,755.7 448.9,759.3 452,762.6 C455.1,765.7 458.9,767.3 463.3,767.3 L559.3,767.3 C563.7,767.3 567.3,765.7 570.6,762.6 C573.8,759.4 575.3,755.6 575.3,751.3 L575.3,751.3 Z"
            />
          </svg>
           
          test
        </li>
        <li
          class=""
        >
          Inspection Scan: sit
        </li>
      </ul>
    </li>
  </ul>
</scansourcelist>
`;

exports[`ScanSourceList Component should render a basic component error: error 1`] = `
<scansourcelist
  props="{
  "id": 1,
  "error": true,
  "errorMessage": "Lorem Ipsum.",
  "pending": false,
  "scanJobList": [],
  "children": []
}"
>
  <div
    class="pf-c-empty-state quipucords-empty-state__alert"
  >
    <div
      class="pf-c-empty-state__content"
    >
      <div
        aria-label="Danger Alert"
        class="pf-c-alert pf-m-inline pf-m-plain pf-m-danger"
        data-ouia-component-id="OUIA-Generated-Alert-danger-1"
        data-ouia-component-type="PF4/Alert"
        data-ouia-safe="true"
      >
        <div
          class="pf-c-alert__icon"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 512 512"
            width="1em"
          >
            <path
              d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
            />
          </svg>
        </div>
        <h4
          class="pf-c-alert__title"
        >
          <span
            class="pf-u-screen-reader"
          >
            Danger alert:
          </span>
          t(view.error, {"context":"scan-jobs"})
        </h4>
        <div
          class="pf-c-alert__description"
        >
          t(view.error-message, {"context":"scan-jobs","message":"Lorem Ipsum."})
        </div>
      </div>
    </div>
  </div>
</scansourcelist>
`;

exports[`ScanSourceList Component should render a basic component pending: pending 1`] = `
<scansourcelist
  props="{
  "id": 1,
  "pending": true,
  "error": false,
  "errorMessage": null,
  "scanJobList": [],
  "children": []
}"
>
  <div
    class="pf-c-empty-state pf-m-lg quipucords-empty-state"
  >
    <div
      class="pf-c-empty-state__content"
    >
      <svg
        aria-label="Contents"
        aria-valuetext="Loading..."
        class="pf-c-spinner pf-m-sm"
        role="progressbar"
        viewBox="0 0 100 100"
      >
        <circle
          class="pf-c-spinner__path"
          cx="50"
          cy="50"
          fill="none"
          r="45"
        />
      </svg>
       
      t(view.loading)
    </div>
  </div>
</scansourcelist>
`;

exports[`ScanSourceList Component should render a basic component: basic 1`] = `
<scansourcelist
  props="{
  "id": 1,
  "scanJobList": [
    {
      "connectTaskStatus": "lorem",
      "connectTaskStatusMessage": "ipsum",
      "id": 1,
      "inspectTaskStatus": "dolor",
      "inspectTaskStatusMessage": "sit",
      "name": "test",
      "sourceType": ""
    }
  ],
  "error": false,
  "errorMessage": null,
  "pending": false,
  "children": []
}"
>
  <ul
    class="pf-c-list pf-m-plain"
  >
    <li
      class=""
    >
      <ul
        class="pf-c-list pf-m-plain"
      >
        <li
          class=""
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 1024 1024"
            width="1em"
          >
            <path
              d="M943.3,317.2 L706.9,80.7 C653,26.9 582.5,0 512,0 C441.5,0 371,26.9 317.1,80.7 L80.7,317.1 C-26.9,424.7 -26.9,599.2 80.7,706.8 L317.1,943.2 C371,997.1 441.5,1024 512,1024 C582.5,1024 653,997.1 706.9,943.3 L943.3,706.9 C1050.9,599.3 1050.9,424.8 943.3,317.2 Z M866.4,602.2 L602.8,866.1 C552.8,916.1 471.8,916.2 421.7,866.2 L157.7,602.2 C107.8,552.2 107.8,471.2 157.8,421.3 L421.4,158.4 C471.3,108.6 552.1,108.6 602.1,158.3 L866.3,421.3 C916.3,471.3 916.3,552.2 866.4,602.2 Z M521.3,576 C627.5,576 713.7,502 713.7,413.7 C713.7,325.4 627.6,253.6 521.3,253.6 C366,253.6 334.5,337.7 329.2,407.2 C329.2,414.3 335.2,416 343.5,416 L445,416 C450.5,416 458,415.5 460.8,406.5 C460.8,362.6 582.9,357.1 582.9,413.6 C582.9,441.9 556.2,470.9 521.3,473 C486.4,475.1 447.3,479.8 447.3,521.7 L447.3,553.8 C447.3,570.8 456.1,576 472,576 C488,576.1 521.3,576 521.3,576 L521.3,576 Z M575.3,751.3 L575.3,655.3 C575.3,650.9 573.7,647.3 570.6,644 C567.5,640.9 563.7,639.3 559.3,639.3 L463.3,639.3 C458.9,639.3 455.3,640.9 452,644 C448.9,647.1 447.3,650.9 447.3,655.3 L447.3,751.3 C447.3,755.7 448.9,759.3 452,762.6 C455.1,765.7 458.9,767.3 463.3,767.3 L559.3,767.3 C563.7,767.3 567.3,765.7 570.6,762.6 C573.8,759.4 575.3,755.6 575.3,751.3 L575.3,751.3 Z"
            />
          </svg>
           
          test
        </li>
        <li
          class=""
        >
          Connection Scan: ipsum
        </li>
      </ul>
    </li>
  </ul>
</scansourcelist>
`;
